src/librustc_borrowck/borrowck/check_loans.rs

```diff
@@ -395,18 +395,29 @@ impl<'a, 'tcx> CheckLoanCtxt<'a, 'tcx> {
         assert!(self.bccx.region_scope_tree.scopes_intersect(old_loan.kill_scope,
                                                        new_loan.kill_scope));
 
-        self.report_error_if_loan_conflicts_with_restriction(
```

Comment:
I am a bit confused on why this actually fixes the error. Maybe this is not the right place to discuss since this was approved already, but shouldn't the old version of report_error_if_loan_conflicts_with_restriction return false if an error is emitted, and thus short-circuiting this if statement (so it is not called the second time)?
Maybe I am missing something here.

Reply:
> Maybe this is not the right place to discuss since this was approved already, but shouldn't the old version of report_error_if_loan_conflicts_with_restriction return false if an error is emitted, and thus short-circuiting this if statement (so it is not called the second time)?

You're right, I believe that was the original intent. I am not sure why short-circuiting wasn't happening.

> p.s. Also why does the the test mention "error: aborting due to 2 previous errors" ? Shouldn't there be only one error?

I think that is because of the cancelled diagnostic, I didn't want to block this PR on that as it is a separate issue. Haven't checked wether there's an issue already, but if not I'll create one.

Reply:
It actually does happen on the current compiler version, I built it and ran the test and it worked as intended.